### PR TITLE
Added MapPanel / Drawer to beta map

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: src/app/pages/MapPage/Components/ViewerContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+`
+
+const ViewerContainer: React.FC<Props> = ({
+  currentOverlay,
+  setCurrentOverlay,
+  showDesktopVariant,
+  setShowDrawTool,
+  setMarkerGroups,
+  markerGroupsRef,
+  isLoading,
```
Comment (jonkoops) [inline src/app/pages/MapPage/Components/ViewerContainer.tsx:68]: The following props are not defined:  

- setShowDrawTool
 - setMarkerGroups
 - markerGroupsRef
 - isLoading

File: src/app/pages/MapPage/Components/ViewerContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          {...otherProps}
+          ignoreTransition={draggable}
+          height={height}
+          topLeft={
+            <BottomLeftHolder>
+              <MapPanelContainerWrapper>
+                <MapPanelContainer />
+              </MapPanelContainerWrapper>
+              <BaseLayerToggle
+                onChangeLayer={(id, type) => console.log(id, type)}
```
Comment (jonkoops) [inline src/app/pages/MapPage/Components/ViewerContainer.tsx:87]: Do we want to keep this console log?

File: src/app/pages/MapPage/MapPage.tsx [hunk 3]
```diff
@@ -1,77 +1,61 @@
...
+  [SnapPoint.Closed]: '30px',
 }
 
 const MapPage: React.FC = () => {
   const [isLoading, setIsLoading] = useState(false)
   const [showDrawTool, setShowDrawTool] = useState(true)
-  const [mapInstance, setMapInstance] = useState<L.Map>()
   const [markerGroups, setMarkerGroups, markerGroupsRef] = useStateRef<MarkerGroup[]>([])
-
+  const [mapInstance, setMapInstance] = useState<L.Map>()
```
Comment (jonkoops) [inline src/app/pages/MapPage/MapPage.tsx:56]: Looks like `mapInstance` is never used here. We can omit it:
```suggestion
  const [, setMapInstance] = useState<L.Map>()
```

File: src/app/pages/MapPage/MapPage.tsx [hunk 4]
```diff
@@ -84,7 +68,19 @@ const MapPage: React.FC = () => {
   } = React.useContext(MapContext)
 
   const tmsLayers = overlays.filter((overlay) => overlay.type === MAP_CONFIG.MAP_LAYER_TYPES.TMS)
-  const nonTmsLayers = overlays.filter((overlay) => overlay.type !== MAP_CONFIG.MAP_LAYER_TYPES.TMS)
+  // const nonTmsLayers = overlays.filter((overlay) => overlay.type !== MAP_CONFIG.MAP_LAYER_TYPES.TMS)
```
Comment (jonkoops) [inline src/app/pages/MapPage/MapPage.tsx:71]: Can we remove this commented out code?

File: src/app/pages/MapPage/MapPage.tsx [hunk 5]
```diff
@@ -94,26 +90,21 @@ const MapPage: React.FC = () => {
...
-          handleMapClick(e, setLocation, setDetailUrl, overlays),
-        )
-      }
-    }
-  }, [mapInstance, overlays])
-
   return (
     <MapView>
+      <Helmet>
+        <meta
```
Comment (jonkoops) [inline src/app/pages/MapPage/MapPage.tsx:96]: Why are we changing the viewport meta?

File: src/app/pages/MapPage/MapPage.tsx [hunk 6]
```diff
@@ -135,43 +125,62 @@ const MapPage: React.FC = () => {
...
-            <BottomLeftHolder>
-              <MapPanelContainerWrapper>
-                <MapPanelContainer />
-              </MapPanelContainerWrapper>
-              <BaseLayerToggle
-                // onChangeLayer={(id, type) => {
-                //   console.log(id, type)
-                // }}
-                activeLayer="luchtfoto"
+        {/*{nonTmsLayers.map(({ url, overlayOptions: options, id }) => (*/}
```
Comment (jonkoops) [inline src/app/pages/MapPage/MapPage.tsx:128]: Can we remove this commented out code?

File: src/app/pages/MapPage/MapContainer.tsx [hunk 7]
```diff
@@ -165,9 +165,10 @@ const MapContextProvider: React.FC<MapContextProps> = ({ children }) => {
...
 
-  function setLocation(payload: Location) {
-    if (payload.lat && payload.lng) {
-      dispatch({ type: 'setLocation', payload })
+  function setLocation(payload: Location | null) {
+    dispatch({ type: 'setLocation', payload })
+
+    if (payload?.lat && payload?.lng) {
       setParams('locatie', encodeLocation(payload))
     }
```
[the remark this answers is not in the document]
Comment (royderks) [reply src/app/pages/MapPage/MapContainer.tsx:173]: @chrisvanmook can you sync with the beta-map branch? Many of these comments have been addresses there yesterday

File: src/app/pages/MapPage/Components/MapLegend.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { mapPanelComponents } from '@datapunt/arm-core'
+
+const { MapPanelContent } = mapPanelComponents
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply src/app/pages/MapPage/Components/MapLegend.tsx:4]: Will change this in te ARM repo

File: src/app/pages/MapPage/Components/MapLegend.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+const MapLegend = ({ ...otherProps }) => (
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply src/app/pages/MapPage/Components/MapLegend.tsx:6]: Will change this in te ARM repo

File: src/app/pages/MapPage/Components/PointSearchMarker.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
+import { LatLng, LeafletMouseEvent } from 'leaflet'
+import { useMapInstance } from '@datapunt/react-maps'
+import { usePanToLatLng, mapPanelComponents, Marker as ARMMarker } from '@datapunt/arm-core'
+import React, { useContext, useEffect } from 'react'
+import { SnapPoint } from './types'
+
+const { MapPanelContext } = mapPanelComponents
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply src/app/pages/MapPage/Components/PointSearchMarker.tsx:7]: Will change this in te ARM repo

File: src/app/pages/MapPage/Components/ViewerContainer.tsx (hunk 2 above)
Comment (chrisvanmook) [reply src/app/pages/MapPage/Components/ViewerContainer.tsx:87]: We still have to bind these to the reducer, I thought I'd leave it in as a reminder but will remove them for now and add a todo

File: src/app/pages/MapPage/MapPage.tsx (hunk 4 above)
Comment (chrisvanmook) [reply src/app/pages/MapPage/MapPage.tsx:71]: Will leave a Todo, there's something going wrong with loading `leaflet.nontiled` in the ARM package

File: src/app/pages/MapPage/MapPage.tsx (hunk 5 above)
Comment (chrisvanmook) [reply src/app/pages/MapPage/MapPage.tsx:96]: Will remove, fixed this in `AppBody` now

**Pull request merged**